test/examples_test.go

```diff
@@ -53,47 +49,23 @@ func getCreatedTektonCRD(input []byte, kind string) (string, error) {
 	return string(submatch[1]), nil
 }
 
-func waitValidatePipelineRunDone(ctx context.Context, t *testing.T, c *clients, pipelineRunName string) {
-	if err := WaitForPipelineRunState(ctx, c, pipelineRunName, pipelineRunTimeout, Succeed(pipelineRunName), pipelineRunName); err != nil {
+func waitValidatePipelineRunDone(ctx context.Context, t *testing.T, c *clients.Clients, pipelineRunName string) {
+	if err := WaitForPipelineRunState(ctx, c.PipelineBetaClient.PipelineRuns, pipelineRunName, pipelineRunTimeout, Succeed(pipelineRunName), pipelineRunName); err != nil {
 		t.Fatalf("Failed waiting for pipeline run done: %v", err)
 	}
 }
 
-func waitValidateTaskRunDone(ctx context.Context, t *testing.T, c *clients, taskRunName string) {
+func waitValidateTaskRunDone(ctx context.Context, t *testing.T, c *clients.Clients, taskRunName string) {
 	// Per test basis
-	if err := WaitForTaskRunState(ctx, c, taskRunName, Succeed(taskRunName), taskRunName); err != nil {
+	if err := WaitForTaskRunState(ctx, c.PipelineBetaClient.TaskRuns, taskRunName, Succeed(taskRunName), taskRunName); err != nil {
 		t.Fatalf("Failed waiting for task run done: %v", err)
 	}
 }
 
-// substituteEnv substitutes docker repos and bucket paths from the system
-// environment for input to allow tests on local clusters. It unsets the
-// namespace for ServiceAccounts so that they work under test. It also
-// replaces image names to arch specific ones, based on provided mapping.
-func substituteEnv(input []byte, namespace string) ([]byte, error) {
-	// Replace the placeholder image repo with the value of the
-	// KO_DOCKER_REPO env var.
-	val, ok := os.LookupEnv("KO_DOCKER_REPO")
-	if !ok {
-		return nil, errors.New("KO_DOCKER_REPO is not set")
-	}
-	output := defaultKoDockerRepoRE.ReplaceAll(input, []byte(val))
-
-	// Strip any "namespace: default"s, all resources will be created in
-	// the test namespace using `ko create -n`
-	output = defaultNamespaceRE.ReplaceAll(output, []byte("namespace: "+namespace))
-
-	// Replace image names to arch specific ones, where it's necessary
-	for existingImage, archSpecificImage := range imagesMappingRE {
-		output = existingImage.ReplaceAll(output, archSpecificImage)
-	}
-	return output, nil
-}
-
 // koCreate wraps the ko binary and invokes `ko create` for input within
 // namespace
 func koCreate(input []byte, namespace string) ([]byte, error) {
-	cmd := exec.Command("ko", "create", "--platform", "linux/"+getTestArch(), "-n", namespace, "-f", "-")
+	cmd := exec.Command("ko", "create", "--platform", testEnv.Platform, "-n", namespace, "-f", "-")
```

Comment:
After test case "initiate tests on amd64, execute on s390x", learnt that `--kubeconfig` should be added here. However it probably can be separate PR, as this functionality is not available in the initial code.

Reply:
@barthy1 I am not sure I understand :thinking:

Reply:
If `go test` is running with `--kubeconfig`, `ko create` won't have knowledge about the special kubeconfig and will be targeted to default cluster

Reply:
@barthy1 ahhh very good point ! Yeah this can be done in a follow-up or now, I'll see if I've got some time to fix